### DISCUSSION TITLE
Query is empty in dynamic route

### DISCUSSION THREAD

**Discussion by nosooqua** [vercel/next.js#23750] [Help]
pages folder:
![image](https://user-images.githubusercontent.com/35642108/113729347-9bd22700-96ff-11eb-8c57-13b944e64041.png)

code: 
![image](https://user-images.githubusercontent.com/35642108/113729438-b2787e00-96ff-11eb-8b76-5125ec004f87.png)

router value:
![image](https://user-images.githubusercontent.com/35642108/113729687-e8b5fd80-96ff-11eb-9bcf-62e78e0e6445.png)

Why query is {}? I have dynamic route [username]/index.js

**Comment by droganov** [comment]
looks similar to https://github.com/vercel/next.js/discussions/23737

the query is only populated after the component mount

**Reply by akellbl4** [reply]
Agree, @nosooqua you should put `username` in the dependency list of  `useEffect`.
